gfx/common/x11_common.c

```diff
@@ -306,9 +306,17 @@ static void x11_suspend_screensaver_xdg_screensaver(Window wnd, bool enable)
 
 void x11_suspend_screensaver(Window wnd, bool enable)
 {
-    x11_suspend_screensaver_xdg_screensaver(wnd, enable);
 #ifdef HAVE_DBUS
     x11_suspend_screensaver_dbus(enable);
+    if (dbus_screensaver_cookie == 0)
+    {
+        // Only attempt xdg-screensaver method if dbus didn't succeed
+#endif
+
+        x11_suspend_screensaver_xdg_screensaver(wnd, enable);
+
+#ifdef HAVE_DBUS
+    }
```

Comment:
Happy to change to something else, I haven't done C in a while so I'm not sure what's idiomatic. I guess I'd prefer duplication over no braces as I found no braces to be error prone, but happy with whatever the style guide is on RetroArch.

Indeed the screensaver is not re-enabled correctly with the xdg-screensaver method (DBus is good I think), that's what I was hinting at in my first comment. Will address separately (this bug was already there).